packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx

```diff
@@ -303,6 +303,12 @@ export class DatePickerBase extends BaseComponent<IDatePickerProps, IDatePickerS
   };
 
   private _onTextFieldBlur = (ev: React.FocusEvent<HTMLElement>): void => {
+    const { onBlur } = this.props;
+
+    if (onBlur) {
```

Comment:
Quick question: if there is any particular reason to raise the event before _validateTextInput?

It might be interesting to document why we raise it in that particular order, since it affects what one can do in the callback. For example, as it is, the callback should know that by the time it is called the selected date and other state might not be their final values.

Would there be a case where we would want the callback to happen after text is parsed and the state is stabilized, ie, call onBlur after _validateTextInput? I'm not sure; arguably, a callback that wants to depend on the new selected date would actually handle onSelectDate.

I just want to make sure we know and understand the implication of the order.

Thanks

Reply:
Fair point, I'll evaluate. If there's a difference I will run it by you and add a unit test (if possible).